### DISCUSSION TITLE
rn firebase phone auth with expo router

### DISCUSSION THREAD

**Discussion by karol-bisztyga** [invertase/react-native-firebase#7686] [Q&A]
Hey!

I'm using expo router and rn firebase authentication via phone number.
When I launch `await auth().signInWithPhoneNumber(phoneNumber)`, it opens Google's verification (captcha), and when I pass the test, it navigates back to the app. The problem is, it doesn't go to the view that opened the captcha, instead, it navigates to `firebaseauth/link`.

I would prefer it to go back to the route where it was called. Do I do something wrong or am I missing something?

Thanks in advance!

**Comment by nihilanthmf** [comment]
Hey! Can you please share your code?

**Comment by rcpankaj** [comment]
Hi @karol-bisztyga,
Try navigating after that function "await auth().signInWithPhoneNumber(phoneNumber)"; your problem will be solved.
After the reCAPTCHA verification, you will be redirected to the screen where you want to go.
